MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h

```diff
@@ -0,0 +1,55 @@
+/** @file MockSmmSwDispatch2.h
+  This file declares a mock of SMM Software Dispatch Protocol
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef MOCK_SMM_SW_DISPATCH2_H_
+#define MOCK_SMM_SW_DISPATCH2_H_
+
+#include <Library/GoogleTestLib.h>
+#include <Library/FunctionMockLib.h>
+
+extern "C" {
+  #include <Uefi.h>
+  #include <Protocol/SmmSwDispatch2.h>
+}
+
+// Declarations to handle usage of the EFI_SMM_SW_DISPATCH2_PROTOCOL
+struct MockSmmSwDispatch2Protocol {
+  MOCK_INTERFACE_DECLARATION (MockSmmSwDispatch2Protocol);
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    MockRegister,
+    (
+     IN  CONST EFI_SMM_SW_DISPATCH2_PROTOCOL  *This,
+     IN        EFI_SMM_HANDLER_ENTRY_POINT2   DispatchFunction,
+     IN  OUT   EFI_SMM_SW_REGISTER_CONTEXT    *RegisterContext,
+     OUT       EFI_HANDLE                     *DispatchHandle)
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    MockUnRegister,
+    (
+     IN CONST EFI_SMM_SW_DISPATCH2_PROTOCOL  *This,
+     IN       EFI_HANDLE                     DispatchHandle)
+    );
+};
+
+MOCK_INTERFACE_DEFINITION (MockSmmSwDispatch2Protocol);
+MOCK_FUNCTION_DEFINITION (MockSmmSwDispatch2Protocol, MockRegister, 4, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockSmmSwDispatch2Protocol, MockUnRegister, 2, EFIAPI);
+
+EFI_SMM_SW_DISPATCH2_PROTOCOL  SMM_SW_DISPATCH2_PROTOCOL_MOCK = {

```

Comment:
Based on the latest discussions in Edk2, the definition of MockEfiSmmSwDispatch2Protocol  should be a #define. Please see the example below:

https://github.com/tianocore/edk2/pull/5950/files#diff-bce28afad342dcfbadd547d3d185dc2159e235b39e204990b683711c22dfc5e9